### PR TITLE
Break Swift CI build into separate job

### PR DESCRIPTION
**Pull request by kevincianfarini**

### CODE REVIEW

**Comment by oldergod** [comment]
This is also what Drafts are for if you wanna try them next time

**Comment by kevincianfarini** [comment]
@oldergod converted to draft!

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -63,7 +63,6 @@ jobs:
         run: |
           sudo xcode-select -switch "/Applications/Xcode_12.app"
```
Comment (JakeWharton) [inline .github/workflows/build.yml:64]: Can be removed

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -77,9 +76,39 @@ jobs:
...
+          java-version: 14
+
+      - name: Run Swift Tests
+        run: |
+          sudo xcode-select -switch "/Applications/Xcode_12.app"
+          sudo gem install cocoapods
+          sudo gem install cocoapods-generate
+          pod gen ./Wire.podspec
+          ./gradlew -p wire-library/wire-runtime-swift build
+          if [ ! -z "$(git status --porcelain)" ]; then git diff; echo -e "\nTest files changed. Did you run ./gradlew generateTests?"; exit 1; fi
```
Comment (JakeWharton) [inline .github/workflows/build.yml:110]: Wrong task in error message

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -77,9 +76,39 @@ jobs:
...
 
+  swift:
+    runs-on: macOS-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Checkout
+        uses: actions/checkout@v2
```
Comment (JakeWharton) [inline .github/workflows/build.yml:90]: Duplicated

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -77,9 +76,39 @@ jobs:
...
+      - name: Configure JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: 14
+
+      - name: Run Swift Tests
+        run: |
+          sudo xcode-select -switch "/Applications/Xcode_12.app"
+          sudo gem install cocoapods
+          sudo gem install cocoapods-generate
```
Comment (JakeWharton) [inline .github/workflows/build.yml:107]: nit: collapse into a single call?

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -61,25 +61,55 @@ jobs:
...
+        # This compiles the golden proto files into Swift files and
+        # ensures that those outputs match what's currently checked into the repo.
+        # These generated files are compiled during the wire-tests-swift build but should still be
+        # reviewed manually.
+        # Finally, run the associated unit tests.
+        run: |
+          sudo xcode-select -switch "/Applications/Xcode_12.app"
+          ./gradlew generateSwiftTests
+          if [ ! -z "$(git status --porcelain)" ]; then git diff; echo -e "\nGenerated files changed. Did you run ./gradlew generateSwiftProtos and generateSwiftTests?"; exit 1; fi
+          ./gradlew -p wire-library :wire-tests-swift:build
```
Comment (JakeWharton) [inline .github/workflows/build.yml:111]: You should only need to do this once
```suggestion
      - run: sudo xcode-select -switch "/Applications/Xcode_12.app"

      - name: Test Swift Runtime
        # This builds the wire-swift-runtime, verifies that generated protos match what's
        # currently checked into the repo both for the runtime itself and for its tests, and runs
        # the tests.
        run: |
          ./gradlew -p wire-library :wire-runtime-swift:build
          if [ ! -z "$(git status --porcelain)" ]; then git diff; echo -e "\nGenerated files changed. Did you run ./gradlew generateSwiftProtos?"; exit 1; fi

      - name: Generate Swift from golden proto files
        # This compiles the golden proto files into Swift files and
        # ensures that those outputs match what's currently checked into the repo.
        # These generated files are compiled during the wire-tests-swift build but should still be
        # reviewed manually.
        # Finally, run the associated unit tests.
        run: |
          ./gradlew generateSwiftTests
          if [ ! -z "$(git status --porcelain)" ]; then git diff; echo -e "\nGenerated files changed. Did you run ./gradlew generateSwiftProtos and generateSwiftTests?"; exit 1; fi
          ./gradlew -p wire-library :wire-tests-swift:build
```

**Review by JakeWharton — APPROVED** [review]

File: .github/workflows/build.yml (hunk 5 above)
Comment (JakeWharton) [reply .github/workflows/build.yml:111]: It just pulls out the xcode-select call. Doesn't really matter...

Comment (kevincianfarini) [reply .github/workflows/build.yml]: yanked it out. Thanks for the sanity check.

**Pull request merged**